src/compiler/scanner.ts

```diff
@@ -2384,15 +2444,979 @@ export function createScanner(languageVersion: ScriptTarget, skipTrivia: boolean
+                                        break;
+                                    }
+                                }
+                                if (max && Number.parseInt(min) > Number.parseInt(max)) {
+                                    error(Diagnostics.Numbers_out_of_order_in_quantifier, digitsStart, pos - digitsStart);
+                                }
+                            }
+                            else if (!min) {
+                                if (unicodeMode) {
+                                    error(Diagnostics.Unexpected_0_Did_you_mean_to_escape_it_with_backslash, start, 1, String.fromCharCode(ch));
+                                }
+                                isPreviousTermQuantifiable = true;
+                                break;
+                            }
+                            scanExpectedChar(CharacterCodes.closeBrace);
+                            pos--;
+                        // falls through
+                        case CharacterCodes.asterisk:
+                        case CharacterCodes.plus:
+                        case CharacterCodes.question:
+                            pos++;
+                            if (text.charCodeAt(pos) === CharacterCodes.question) {
+                                // Non-greedy
+                                pos++;
+                            }
+                            if (!isPreviousTermQuantifiable) {
+                                error(Diagnostics.There_is_nothing_available_for_repetition, start, pos - start);
+                            }
+                            isPreviousTermQuantifiable = false;
+                            break;
+                        case CharacterCodes.dot:
+                            pos++;
+                            isPreviousTermQuantifiable = true;
+                            break;
+                        case CharacterCodes.openBracket:
+                            pos++;
+                            if (unicodeSetsMode) {
+                                scanClassSetExpression();
+                            }
+                            else {
+                                scanClassRanges();
+                            }
+                            scanExpectedChar(CharacterCodes.closeBracket);
+                            isPreviousTermQuantifiable = true;
+                            break;
+                        case CharacterCodes.closeParen:
+                            if (isInGroup) {
+                                return;
+                            }
+                        // falls through
+                        case CharacterCodes.closeBracket:
+                        case CharacterCodes.closeBrace:
+                            if (isUnterminated && !isInGroup) {
+                                // Assume what starting from the character to be outside of the regex
+                                return;
+                            }

```

Comment:
@rbuckton
> Performing a full parse of a RegExp during scan seems like unnecessary overhead for those scenarios.

Unfortunately `scanRegularExpressionWorker` may change the end position of a regex due to this part. See the change in `src/testRunner/unittests/incrementalParser.ts` for an example.

(`!isInGroup` indicates we are currently at the outermost level, all open parentheses are closed)

Reply:
I’ll try to take this out independently when I move the worker to the parser.